kuka_rox_hardware_interface/src/mock/os-core-udp-communication/udp_replier_mock.cpp

```diff
@@ -6,46 +6,38 @@
 // Copyright (C)
 // KUKA Deutschland GmbH, Germany. All Rights Reserved.
 
-#include "os-core-udp-communication/udp_replier.h"
+#include "os-core-udp-communication/replier.h"
 
 namespace os::core::udp::communication
 {
-
-UDPReplier::UDPReplier(const SocketAddress & local_address)
-: local_address_(local_address) {}
-
-UDPSocket::ErrorCode UDPReplier::Setup()
+Replier::Replier(const SocketAddress & local_address)
 {
   // only a mock
-  return UDPSocket::ErrorCode::kSuccess;
 }
-
-void UDPReplier::Reset()
+Socket::ErrorCode Replier::Setup()
 {
   // only a mock
 }
-
-UDPSocket::ErrorCode UDPReplier::ReceiveRequest()
+void Replier::Reset()
 {
   // only a mock
-  return UDPSocket::ErrorCode::kSuccess;
 }
 
-UDPSocket::ErrorCode UDPReplier::ReceiveRequestOrTimeout(std::chrono::microseconds recv_timeout)
+Socket::ErrorCode Replier::ReceiveRequest()
 {
   // only a mock
-  return UDPSocket::ErrorCode::kSuccess;
 }
-
-UDPSocket::ErrorCode UDPReplier::SendReply(uint8_t * reply_msg_data, size_t reply_msg_size)
+Socket::ErrorCode Replier::ReceiveRequestOrTimeout(std::chrono::microseconds recv_timeout)
 {
   // only a mock
-  return UDPSocket::ErrorCode::kSuccess;
 }
-
-std::pair<const uint8_t *, size_t> UDPReplier::GetRequestMessage() const
+Socket::ErrorCode Replier::SendReply(uint8_t * reply_msg_data, size_t reply_msg_size)
 {
-  return {nullptr, 0};
+  // only a mock
 }
 
+std::pair<const uint8_t *, size_t> Replier::GetRequestMessage() const
+{
```

Comment:
Even though this is "only a mock",  missing return statements are undefined behavior. You should return some garbage value here.